src/transformers/keras_callbacks.py

```diff
@@ -69,13 +71,24 @@ def __init__(
         if "/" not in hub_model_id:
             hub_model_id = get_full_repo_name(hub_model_id, token=hub_token)
 
+        # TODO Is it okay to store the hub token as a callback attribute like this?
+        self.hub_token = hub_token
```

Comment:
If you store it, make sure it doesn't appear in the repr (we had that surprise in the TrainingArguments) or something saved automatically.

Reply:
Is there any way to make this work without storing the hub_token? I could move the repo creation to `__init__`, but that might surprise users when the repo is made when the callback is made, rather than when training begins.

Reply:
I think it's better to do it at init and have a potential error sooner rather than later.

Reply:
I set it to clear the `hub_token` attribute as soon as the repo is created, and made sure it wasn't being printed in the `__repr__`, so I think this is fairly safe.

Reply:
But why not create the repo at init? Is there some infor we only get later?

Reply:
The main reason is that it's surprising for Keras users - Callbacks usually don't do anything visible at init, because they exist to trigger actions at certain points during training. If you think the problems with moving initialization to `on_train_begin` are too annoying, though, I can create the repo there, I'm sure it's fine!

Reply:
In this case I find it weird to not do it at init (the Trainer does it there FYI): this method clones the repo where we will work, which is something you do before you execute your script in normal life. If there is an error because the repo already exists and doesn't match the `clone_from` it will be more obvious if the error is raised at the callback creation rather when launching the training.

Reply:
That makes sense, yes. I'll move it!

Reply:
Done!